src/enclave/enclave_signal.c

```diff
@@ -143,6 +143,32 @@ static uint64_t sgxlkl_enclave_signal_handler(
             exception_record->address,
             opcode);
 
+        /**
+         * If LKL has not yet been initialised, we cannot handle the
+         * exception and fail instead.
+         */
+        if (!lkl_is_running())
+        {
+#ifdef DEBUG
+            sgxlkl_error("Unhandled exception. Printing stack trace before the signal handler was invoked:\n");
+            sgxlkl_print_backtrace((void*)oe_ctx->rbp);
+#endif
+
+            struct lthread* lt = lthread_self();
+            sgxlkl_fail(
+                "Exception %s received before LKL is running (lt->tid=%i [%s] "
+                "code=%i "
+                "addr=0x%lx opcode=0x%x "
+                "ret=%i)\n",
+                trap_info.description,
+                lt ? lt->tid : -1,
+                lt ? lt->funcname : "(?)",
+                exception_record->code,
+                (void*)exception_record->address,
+                opcode,
+                ret);
+        }
+
         memset(&uctx, 0, sizeof(uctx));
```

Comment:
I think for auditing it would be nice if there was a comment saying where unhandled exceptions are processed in case LKL *is* running. The code below is likely hard to follow to the final location.

Reply:
Done